### DISCUSSION TITLE
Text replacement

### DISCUSSION THREAD

**Discussion by Tangerino** [qpdf/qpdf#1252] [General]
I use a lot of PDF files to display energy bill information. I need to create some fake bills, by masking, for example, the customer name, address, etc.
So the idea is to grab an original pdf bill, mask/replace some fields and write back a new pdf that will be used for demo.
Is that possible?
Thank you in advance

**Comment by jberkenbilt** [comment]
qpdf won't be the right tool for that kind of job. You'll need something capable of editing the content of PDF files. You'll need to be careful though -- simply applying masks on top of something will not remove the underlying text. The text you want to remove needs to be fully removed. I'm not specifically aware of tools that do this, though I know many exist. I don't know if any are open source. Just make sure they are doing true redaction, not just covering things up.

**Comment by m-holger** [comment]
You could try editing the PDF with LibreOffice. That usually works for me.

**Reply by m-holger** [reply]
and export the edited file as PDF.
